clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -705,6 +726,18 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::init(
 {
     options_ = std::move(options);
     verbosity_ = options_.verbosity;
+
+    // Add noise generator reader to workload_inputs.
+    if (options_.enable_noise_generator) {
+        auto noise_generator = get_noise_generator(options_.noise_generator_num_records);
+        auto noise_generator_end = get_noise_generator(0);
```

Comment:
Perhaps we should have a separate `get_noise_generator_end()`?

Reply:
Agreed.

Reply:
Done.